.github/workflows/lint.yml

```diff
@@ -0,0 +1,85 @@
+name: lint
+
+on:
+  push:
+    branches:
+      - main
+      - 'release/**'
+  pull_request:
+
+env:
+  GO_VERSION: 1.23.x
+
+jobs:
+  go:
+    timeout-minutes: 5
+    name: "go | ${{ matrix.goos }} | ${{ matrix.gonext }}"
+    runs-on: "${{ matrix.os }}"
+    defaults:
+      run:
+        shell: bash
+    strategy:
+      matrix:
+        include:
+          - os: ubuntu-24.04
+            goos: linux
+          - os: ubuntu-24.04
+            goos: freebsd
+          # FIXME: this is currently failing in a non-sensical way, so, running on linux instead...
+          # - os: windows-2022
+          - os: ubuntu-24.04
+            goos: windows
+          - os: ubuntu-24.04
+            goos: linux
+            # This allows the canary script to select any upcoming golang alpha/beta/RC
+            gonext: canary
```

Comment:
```suggestion
            canary: true
```

Reply:
Definitely better. Thanks.

Wondering about the github UI: can I just accept "signoff and commit suggestion" or is it going to create an extraneous commit?

Reply:
>  is it going to create an extraneous commit?

Yes, and the web UI doesn't seem to have the "squash" button  😞